library/core/src/panic.rs

```diff
@@ -58,6 +58,42 @@ pub macro panic_2021 {
     ),
 }
 
+#[doc(hidden)]
+#[unstable(feature = "edition_panic", issue = "none", reason = "use unreachable!() instead")]
+#[allow_internal_unstable(core_panic, const_format_args)]
+#[rustc_diagnostic_item = "unreachable_2015_macro"]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro unreachable_2015 {
+    () => (
+        $crate::panicking::panic("internal error: entered unreachable code")
+    ),
+    // Use of `unreachable_display` for non_fmt_panic lint.
+    // NOTE: the message ("internal error ...") is embeded directly in unreachable_display
+    ($msg:expr $(,)?) => (
+        $crate::panicking::unreachable_display(&$msg)
+    ),
+    ($fmt:expr, $($arg:tt)*) => (
+        $crate::panic!($crate::concat!("internal error: entered unreachable code: ", $fmt), $($arg)*)
+    ),
+}
+
+#[doc(hidden)]
+#[unstable(feature = "edition_panic", issue = "none", reason = "use unreachable!() instead")]
+#[allow_internal_unstable(core_panic, const_format_args)]
+#[rustc_diagnostic_item = "unreachable_2021_macro"]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro unreachable_2021 {
+    () => (
+        $crate::panicking::panic("internal error: entered unreachable code")
+    ),
+    ($($t:tt)+) => (
+        $crate::panicking::panic_fmt(
+            $crate::const_format_args!(
+                "internal error: entered unreachable code: {}",
+                    $crate::const_format_args!($($t)+)))
+    ),
+}
```

Comment:
Let's just use the `panic!()` macro in the second case, to follow the `todo` and `unimplemented` macros: https://github.com/rust-lang/rust/blob/a34c0797528172ede89480e3033f7a5e71ea4735/library/core/src/macros/mod.rs#L688-L691

There's no need for the `const_format_args` here. The regular `format_args` should suffice, as none of this would work in const context anyway.